docs/admin/node.md

```diff
@@ -120,25 +120,48 @@ Currently, there are three components that interact with the Kubernetes node int
 ### Node Controller
 
 Node controller is a component in Kubernetes master which manages Node
-objects. It performs two major functions: cluster-wide node synchronization
-and single node life-cycle management.
-
-Node controller has a sync loop that deletes Nodes from Kubernetes
-based on all matching VM instances listed from the cloud provider. The sync period
-can be controlled via flag `--node-sync-period`. If a new VM instance
-gets created, Node Controller creates a representation for it. If an existing
-instance gets deleted, Node Controller deletes the representation. Note however,
-that Node Controller is unable to provision the node for you, i.e. it won't install
-any binary; therefore, to
-join a node to a Kubernetes cluster, you as an admin need to make sure proper services are
-running in the node. In the future, we plan to automatically provision some node
-services.
-
-In general, node controller is responsible for updating the NodeReady condition of node
-status to ConditionUnknown when a node becomes unreachable (e.g. due to the node being down),
-and then later evicting all the pods from the node (using graceful termination) if the node
-continues to be unreachable. (The current timeouts for those are 40s and 5m, respectively.)
-It also allocates CIDR blocks to the new nodes.
+objects.
+
+Node controller has mutliple roles in Node's life. First is assigning a CIDR block to
+the Node when it is registered (if CIDR assignment is turned on). Second part is monitoring
+Node's health. Node controller is responsible for updating the NodeReady condition of node
+status to ConditionUnknown when a node becomes unreachable (i.e. Kubelet stopped sending heartbeats
+e.g. due to the node being down), and then later evicting all the pods from the node
+(using graceful termination) if the node continues to be unreachable (the current timeouts
+for those are 40s and 5m, respectively). Node controller checks the state of each node every
+`--node-monitor-period` seconds.
+
+In 1.4 release we updated the logic of node controller to better handle cases when a
+big number of Nodes have problems with reaching the master machine (e.g. because
+master machine has networking problem). Since 1.4 node controller will look at the
```

Comment:
s/Since/Starting with/

I would change the "In previous releases..." sentence a bit; I think it's a little clearer to say "In previous releases if the master machine got partitioned from the cluster for at least 5m40s, it would delete all running pods from its "desired state" for the cluster, and then when the partition ended and the kubelets were able to contact the master again, they would see all of their pods had been deleted and would kill them."